extensions/standard-processors/processors/GenerateFlowFile.cpp

```diff
@@ -77,7 +73,28 @@ void GenerateFlowFile::initialize() {
   setSupportedRelationships(relationships);
 }
 
+void generateData(std::vector<char>& data, bool textData = false) {
+  std::random_device rd;
+  std::mt19937 eng(rd());
+  if (textData) {
+    std::uniform_int_distribution<> distr(0, TEXT_LEN);
+    auto rand = std::bind(distr, eng);
+    std::generate_n(data.begin(), data.size(), rand);
+    std::for_each(data.begin(), data.end(), [](char & c) { c = TEXT_CHARS[c];});
+  } else {
+    std::uniform_int_distribution<> distr(std::numeric_limits<char>::min(), std::numeric_limits<char>::max());
+    auto rand = std::bind(distr, eng);
+    std::generate_n(data.begin(), data.size(), rand);
+  }
+}
+
 void GenerateFlowFile::onTrigger(core::ProcessContext *context, core::ProcessSession *session) {
+  std::unique_lock<std::mutex> lock(onTriggerMutex_, std::try_to_lock);
+  if (!lock.owns_lock()) {
+    logger_->log_warn("processor was triggered before previous listing finished, configuration should be revised!");
```

Comment:
This error message seems like it applies elsewhere since we are generating data, not listing.

Reply:
Removed completely as it became needless.